cmd/es-rollover/app/init/action_test.go

```diff
@@ -18,60 +17,82 @@ import (
 )
 
 func TestIndexCreateIfNotExist(t *testing.T) {
-	const esErrResponse = `{"error":{"root_cause":[{"type":"resource_already_exists_exception","reason":"]"}],"type":"resource_already_exists_exception","reason":"request [/jaeger-*] contains unrecognized parameter: [help]"},"status":400}`
-
+	t.Run("index exists", func(t *testing.T) {
+		testCreateIndexIfNotExist(t, "index")
+	})
+	t.Run("alias exists", func(t *testing.T) {
+		testCreateIndexIfNotExist(t, "alias")
+	})
 	tests := []struct {
-		name          string
-		returnErr     error
-		expectedErr   error
-		containsError string
+		name                     string
+		createIndexReturnErr     error
+		createIndexExpectedError error
+		createIndexContainsError string
 	}{
 		{
-			name: "success",
+			name:                     "generic error from create index",
+			createIndexReturnErr:     errors.New("may be an http error from create index"),
+			createIndexExpectedError: errors.New("may be an http error from create index"),
 		},
 		{
-			name:        "generic error",
-			returnErr:   errors.New("may be an http error?"),
-			expectedErr: errors.New("may be an http error?"),
+			name:                     "response error",
+			createIndexReturnErr:     errors.New("x"),
+			createIndexExpectedError: errors.New("x"),
 		},
 		{
-			name: "response error",
-			returnErr: client.ResponseError{
-				Err:        errors.New("x"),
-				StatusCode: http.StatusForbidden,
-			},
-			expectedErr: errors.New("x"),
+			name: "success",
 		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			indexClient := &mocks.IndexAPI{}
+			indexClient.On("IndexExists", "jaeger-span").Return(false, nil)
+			indexClient.On("AliasExists", "jaeger-span").Return(false, nil)
+			indexClient.On("CreateIndex", "jaeger-span").Return(test.createIndexReturnErr)
+			err := createIndexIfNotExist(indexClient, "jaeger-span")
+			if test.createIndexContainsError != "" {
+				assert.ErrorContains(t, err, test.createIndexContainsError)
+			} else {
+				assert.Equal(t, test.createIndexExpectedError, err)
+			}
+		})
+	}
+}
+
+func testCreateIndexIfNotExist(t *testing.T, existence string) {
```

Comment:
this is odd test organization. You can make this public Testxxx instead of calling from elsewhere, and the distinction between index/alias parameters can be encoded in the data matrix below, e.g.
```
struct {
		name                string
		indexExists         bool
		indexExistsErr      error
		aliasExists         bool
		aliasExistsErr      error
		expectedError       string
	}
```